irohad/ametsuchi/impl/postgres_setting_query.cpp

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ametsuchi/impl/postgres_setting_query.hpp"
+
+#include <boost/lexical_cast.hpp>
+#include "interfaces/common_objects/types.hpp"
+#include "logger/logger.hpp"
+
+using namespace iroha::ametsuchi;
+
+namespace {
+  template <typename T>
+  bool getValueFromDb(soci::session &sql,
+                      const shared_model::interface::types::SettingKeyType &key,
+                      T &destination) {
+    boost::optional<shared_model::interface::types::SettingValueType> value;
+
+    sql << "SELECT setting_value FROM setting WHERE setting_key = :key",
+        soci::into(value), soci::use(key, "key");
+
+    if (value) {
+      destination = boost::lexical_cast<T>(value.get());
+      return true;
+    }
+    return false;
+  }
+}  // namespace
+
+PostgresSettingQuery::PostgresSettingQuery(std::unique_ptr<soci::session> sql,
+                                           logger::LoggerPtr log)
+    : psql_(std::move(sql)), sql_(*psql_), log_(std::move(log)) {}
+
+iroha::expected::Result<
+    std::shared_ptr<const shared_model::validation::Settings>,
+    std::string>
+PostgresSettingQuery::get() {
+  std::shared_ptr<shared_model::validation::Settings> default_settings =
+      shared_model::validation::getDefaultSettings();
+  return update(default_settings);
+}
+
+iroha::expected::Result<
+    std::shared_ptr<const shared_model::validation::Settings>,
+    std::string>
+PostgresSettingQuery::update(
+    std::shared_ptr<shared_model::validation::Settings> base) {
+  auto get_and_log =
+      [this](const shared_model::interface::types::SettingKeyType &key,
+             auto &destination) {
+        if (getValueFromDb(sql_, key, destination)) {
+          log_->info("Updated value for " + key + ": {}", destination);
+        } else {
+          log_->info("Kept value for " + key + ": {}", destination);
+        }
+      };
+
+  try {
+    get_and_log(shared_model::validation::Settings::kMaxDescriptionSizeKey,
```

Comment:
lambda seems not needed here.
```suggestion
    if (getValueFromDb(
            sql_,
            hared_model::validation::Settings::kMaxDescriptionSizeKey,
            base->max_description_size)) {
      log_->info("Updated value for " + key + ": {}", destination);
    } else {
      log_->info("Kept value for " + key + ": {}", destination);
    }
```

Reply:
But when adding more settings, there will be a lot of code

Reply:
lambda does not solve this problem.

Reply:
Maybe like this, otherwise for adding some new setting need to write one more if statement
```C++
template <typename T>
void getAndLog(const shared_model::interface::types::SettingKeyType &key,
               T &destination,
               soci::session &sql,
               const logger::LoggerPtr& log) {
  if (getValueFromDb(sql, key, destination)) {
    log->info("Updated value for " + key + ": {}", destination);
  } else {
    log->info("Kept value for " + key + ": {}", destination);
  }
}
```

Reply:
now I got the meaning of lambda, sorry for confusing you. lambda is good here.